pkgs/build-support/dotnet/add-nuget-deps/default.nix

```diff
@@ -20,16 +47,17 @@ let
   attrs = if builtins.isFunction fnOrAttrs then fnOrAttrs finalAttrs else fnOrAttrs;
 
   deps =
-    if (nugetDeps != null) then
-      if lib.isDerivation nugetDeps then
-        [ nugetDeps ]
-      else if lib.isList nugetDeps then
-        nugetDeps
-      else
-        assert (lib.isPath nugetDeps);
-        callPackage nugetDeps { fetchNuGet = fetchNupkg; }
+    if nugetDeps == null then
+      [ ]
+    else if lib.isDerivation nugetDeps then
+      [ nugetDeps ]
+    else if lib.isList nugetDeps then
+      nugetDeps
+    else if lib.hasSuffix ".nix" nugetDeps then
+      assert (lib.isPath nugetDeps);
+      callPackage nugetDeps { fetchNuGet = fetchNupkg; }
     else
-      [ ];
+      builtins.map fetchNupkg (lib.importJSON nugetDeps);
```

Comment:
Same re: assert position

Reply:
The assert is there for legacy reasons really, since it was there for the nix path I just left it there, but I don't think it is really needed honestly.